rpcs3/Emu/RSX/VK/VKFragmentProgram.cpp

```diff
@@ -228,43 +228,17 @@ void VKFragmentDecompilerThread::insertMainStart(std::stringstream & OS)
 		{
 			if (m_prog.front_back_color_enabled)
 			{
-				if (PI.name == "spec_color")
+				if (PI.name == "spec_color" && m_prog.back_color_specular_output && m_prog.front_color_specular_output)
```

Comment:
It will generate shader error if proposed condition used .

```
E {rsx::thread} RSX: Warning, version 420 is not yet complete; most version-specific features are present, but some are missing.
ERROR: 0:65: 'back_diff_color' : undeclared identifier 
ERROR: 0:65: ':' :  wrong operand types: no operation ':' exists that takes a left-hand operand of type 'layout(location=11 ) smooth in 4-component vector of float' and a right operand of type 'temp float' (or there is no acceptable conversion)
ERROR: 0:65: '=' :  cannot convert from 'temp float' to 'temp 4-component vector of float'
ERROR: 3 compilation errors.  No code generated.

```